app/src/main/java/com/example/sridharjajoo/newsapp/core/Search/SearchFragment.java

```diff
@@ -133,10 +136,17 @@ private void handleSearch(String query) {
         searchViewModel.customSearch(query).observe(this, this::showResult);
     }
 
+
+
     private void showResult(List<Articles> articles) {
         if (articles == null) {
             return;
         }
+        if(articles.size()==0)
```

Comment:
remove all the extra lines in this class and couple this if with the above one as if (articles == null || articles.size( ) == 0)

Reply:
Also suppose we search for a result and we get it and then we search for another query to which no response is received then it that case the old recycler view and the No Search result placeholder will overlap. So after every query when a new query is searched for empty the current recycler view